### PR TITLE
#5994 refactor rename typescript types

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Part of #5994 

## Tips for Reviewers
- I called out a handful of type changes that weren't explicitly mentioned in the spec

## Discussion

- I'll be adding more changes, just getting the Draft PR out for visibility

## Future Work

- Continue working through type renaming

## Checklist

- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

**Review by twschiller — APPROVED** [review]

**Review by BLoe — DISMISSED** [review]

File: src/extensionPoints/types.ts [hunk 1]
```diff
@@ -88,35 +88,33 @@ export interface ExtensionPointConfig<
...
 export function assertExtensionPointConfig(
   maybeExtensionPointConfig: unknown
-): asserts maybeExtensionPointConfig is ExtensionPointConfig {
+): asserts maybeExtensionPointConfig is StarterBrickConfig {
   const errorContext = { value: maybeExtensionPointConfig };
 
   if (typeof maybeExtensionPointConfig !== "object") {
     console.warn("Expected extension point", errorContext);
-    throw new TypeError("Expected object for ExtensionPointConfig");
+    throw new TypeError("Expected object for StarterBrickConfig");
```
Comment (grahamlangford) [inline src/extensionPoints/types.ts:96]: Is there any risk involved with this change? Will it impact anything in MixPanel or Rollbar?

File: src/recipes/registry.ts [hunk 2]
```diff
@@ -35,11 +35,11 @@ type UnnormalizedOptionsDefinition = {
   uiSchema?: UiSchema;
 };
 
-type UnnormalizedRecipeDefinition = Exclude<ModDefinition, "options"> & {
+type UnnormalizedModDefinition = Exclude<ModDefinition, "options"> & {
```
Comment (grahamlangford) [inline src/recipes/registry.ts:38]: Calling this change out because it wasn't included in the spec.

File: src/extensionPoints/types.ts [hunk 3]
```diff
@@ -71,14 +71,14 @@ export type CustomEventOptions = {
   eventName: string;
 };
 
-export interface ExtensionPointDefinition {
+export interface StarterBrickDefinition {
```
Comment (grahamlangford) [inline src/extensionPoints/types.ts:74]: Calling this change out because it wasn't included in the spec.

File: src/extensionPoints/types.ts [hunk 4]
```diff
@@ -71,14 +71,14 @@ export type CustomEventOptions = {
   eventName: string;
 };
 
-export interface ExtensionPointDefinition {
-  type: ExtensionPointType;
+export interface StarterBrickDefinition {
+  type: StarterBrickType;
```
Comment (grahamlangford) [inline src/extensionPoints/types.ts:75]: Calling this change out because it wasn't included in the spec.

File: src/extensionPoints/types.ts (hunk 3 above)
Comment (twschiller) [reply src/extensionPoints/types.ts:74]: LGTM

File: src/extensionPoints/types.ts (hunk 1 above)
Comment (twschiller) [reply src/extensionPoints/types.ts:96]: Nope, error messages should be safe

**Pull request merged**